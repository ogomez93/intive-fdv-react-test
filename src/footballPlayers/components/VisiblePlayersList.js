import React from 'react';
import { connect } from 'react-redux';
import Player from './Player';
import { getVisiblePlayers } from '../selectors';

const VisiblePlayersList = ({ players }) => (
  <tbody>
    {
      !players.length ? 
        <tr id="noPlayersMatchFilters">
          <td className="border-2 border-black px-1 py-3 sm:py-6" colSpan="4">No players match those filters</td>
        </tr>
      :
        players.map(player => <Player player={player} key={player.name} />)
    }
  </tbody>
);

const mapStateToProps = (state) => {
  return {
    players: getVisiblePlayers(state)
  }
}

export default connect(
  mapStateToProps
)(VisiblePlayersList);