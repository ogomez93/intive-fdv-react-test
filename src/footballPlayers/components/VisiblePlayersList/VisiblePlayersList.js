import React from 'react';
import Player from 'footballPlayers/components/Player';

const VisiblePlayersList = ({ players }) => (
  <tbody>
    { players.map(player => <Player {...player} key={player.name} />) }
  </tbody>
);

export default VisiblePlayersList
