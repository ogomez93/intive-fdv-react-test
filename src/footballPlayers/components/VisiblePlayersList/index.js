import { connect } from 'react-redux';
import { branch, compose, renderComponent } from 'recompose';
import VisiblePlayersList from './VisiblePlayersList';
import NoPlayers from 'footballPlayers/components/NoPlayers';
import { getVisiblePlayers } from 'footballPlayers/selectors';

const mapStateToProps = state => ({
  players: getVisiblePlayers(state)
});

export default compose(
  connect(mapStateToProps),
  branch(({ players }) => players.length === 0, renderComponent(NoPlayers))
)(VisiblePlayersList);
