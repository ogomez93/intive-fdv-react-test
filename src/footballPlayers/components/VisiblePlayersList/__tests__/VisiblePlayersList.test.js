import React from 'react';
import { shallow } from 'enzyme';

import VisiblePlayersList from '../VisiblePlayersList';

import props from './fixtures/players';

jest.mock('../../Player', () => 'Player');

describe('VisiblePlayersList', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<VisiblePlayersList {...props} />);
  });
  
  it('should render', () => {
    expect(wrapper.isEmptyRender()).toBe(false);
  });
  
  it('should render two players', () => {
    expect(wrapper.prop('children').length).toEqual(2);
  });
  
  describe('player 1', () => {
    let player1;

    beforeAll(() => {
      player1 = wrapper.find('Player').at(0);
    });
    
    it('should be a Player component', () => {
      expect(player1.type()).toEqual('Player');
    });

    it('should have the right props', () => {
      expect(player1.props()).toMatchSnapshot();
    });
  });
  
  describe('player 2', () => {
    let player2;

    beforeAll(() => {
      player2 = wrapper.find('Player').at(1);
    });
    
    it('should be a Player component', () => {
      expect(player2.type()).toEqual('Player');
    });

    it('should have the right props', () => {
      expect(player2.props()).toMatchSnapshot();
    });
  });
});
