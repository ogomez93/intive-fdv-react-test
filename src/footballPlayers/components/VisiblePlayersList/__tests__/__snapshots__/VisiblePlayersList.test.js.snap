// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisiblePlayersList player 1 should have the right props 1`] = `
Object {
  "dateOfBirth": "player 1 dateOfBirth",
  "name": "player 1 name",
  "nationality": "player 1 nationality",
  "position": "player 1 position",
}
`;

exports[`VisiblePlayersList player 2 should have the right props 1`] = `
Object {
  "dateOfBirth": "player 2 dateOfBirth",
  "name": "player 2 name",
  "nationality": "player 2 nationality",
  "position": "player 2 position",
}
`;
